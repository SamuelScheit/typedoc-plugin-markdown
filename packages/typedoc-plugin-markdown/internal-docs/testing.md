---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
