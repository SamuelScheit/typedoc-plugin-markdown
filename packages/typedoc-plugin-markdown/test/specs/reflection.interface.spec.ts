import { expectFileToEqual } from '@devtools/testing';

describe(`Interface Reflection`, () => {
  test(`should compile basic interface`, () => {
    expectFileToEqual('reflections', 'members', 'interfaces/BasicInterface.md');
  });

  test(`should compile interface with type parameters`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'interfaces/InterfaceWithTypeParameters.md',
    );
  });

  test(`should compile extended interface`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'interfaces/ExtendedInterface.md',
    );
  });

  test(`should compile indexable interface`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'interfaces/IndexableInterface.md',
    );
  });

  test(`should compile interface with event properties`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'interfaces/InterfaceWithEventProperties.md',
    );
  });
});
