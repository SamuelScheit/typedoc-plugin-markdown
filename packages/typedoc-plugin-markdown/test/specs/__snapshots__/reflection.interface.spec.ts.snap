// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interface Reflection should compile basic interface: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Interface: BasicInterface

Comments for BasicInterface

## Extended By

- [\`ExtendedInterface\`](ExtendedInterface.md)

## Contents

- [Properties](BasicInterface.md#properties)
    - [deprecatedProp](BasicInterface.md#deprecatedprop)
    - [functionProp](BasicInterface.md#functionprop)
    - [optionalProp](BasicInterface.md#optionalprop)
    - [prop](BasicInterface.md#prop)
    - [propWithProps](BasicInterface.md#propwithprops)

## Properties

### ~~deprecatedProp~~

> **deprecatedProp**: \`string\`

#### Deprecated

This prop is deprecte

#### Some Tag

Comments for some tag

#### Source

[interfaces.ts:25](http://source-url)

***

### functionProp

> **functionProp**: (\`s\`) => \`boolean\`

Comments for functionProper

#### Parameters

• **s**: \`string\`

Comment for param s

#### Returns

\`boolean\`

#### Source

[interfaces.ts:36](http://source-url)

***

### optionalProp?

> **optionalProp**?: \`string\`

Comments for optional prop

#### Source

[interfaces.ts:30](http://source-url)

***

### prop

> **prop**: \`string\`

Comments for prop

#### Source

[interfaces.ts:14](http://source-url)

***

### propWithProps

> **propWithProps**: \`Object\`

Comments for propWithProps

#### Type declaration

##### callbacks?

> **callbacks**?: \`Partial\`\\<[\`CallbacksOptions\`](../classes/CallbacksOptions.md)\\<[\`DisposableClass\`](../classes/DisposableClass.md), [\`ClassWithModifiers\`](../classes/ClassWithModifiers.md)\\>\\>

Comments for callbacks

##### nestedPropA

> **nestedPropA**: \`string\`

Comments for nestedPropA

##### nestedPropB

> **nestedPropB**: \`boolean\`

Comments for nestedPropB

##### nestedPropC

> **nestedPropC**: \`Object\`

Comments for nestedPropC

##### nestedPropC.nestedPropCA

> **nestedPropC.nestedPropCA**: \`string\`

Comments for nestedPropCA

##### nestedPropD

> **nestedPropD**: () => \`boolean\`

Comments for nestedPropD

###### Returns

\`boolean\`

#### Source

[interfaces.ts:41](http://source-url)
"
`;

exports[`Interface Reflection should compile basic interface: (Output File Strategy "members") (Option Group "2") 1`] = `
"# BasicInterface

Comments for BasicInterface

## Extended By

- [\`ExtendedInterface\`](ExtendedInterface.md)

## Contents

- [Properties](BasicInterface.md#properties)

## Properties

| Property | Type | Description |
| :------ | :------ | :------ |
| ~~\`deprecatedProp\`~~ | \`string\` | **Deprecated**<br />This prop is deprecte<br /><br />**Some Tag**<br />Comments for some tag |
| \`functionProp\` | (\`s\`) => \`boolean\` | - |
| \`optionalProp?\` | \`string\` | Comments for optional prop |
| \`prop\` | \`string\` | Comments for prop |
| \`propWithProps\` | \\{   \`callbacks\`: \`Partial\`\\<[\`CallbacksOptions\`](../classes/CallbacksOptions.md)\\<[\`DisposableClass\`](../classes/DisposableClass.md), [\`ClassWithModifiers\`](../classes/ClassWithModifiers.md)\\>\\>;   \`nestedPropA\`: \`string\`;   \`nestedPropB\`: \`boolean\`;   \`nestedPropC\`: \\{      \`nestedPropCA\`: \`string\`;   };   \`nestedPropD\`: () => \`boolean\`;   } | Comments for propWithProps |
| \`propWithProps.callbacks?\` | \`Partial\`\\<[\`CallbacksOptions\`](../classes/CallbacksOptions.md)\\<[\`DisposableClass\`](../classes/DisposableClass.md), [\`ClassWithModifiers\`](../classes/ClassWithModifiers.md)\\>\\> | Comments for callbacks |
| \`propWithProps.nestedPropA\` | \`string\` | Comments for nestedPropA |
| \`propWithProps.nestedPropB\` | \`boolean\` | Comments for nestedPropB |
| \`propWithProps.nestedPropC\` | \\{   \`nestedPropCA\`: \`string\`;   } | Comments for nestedPropC |
| \`propWithProps.nestedPropC.nestedPropCA\` | \`string\` | Comments for nestedPropCA |
| \`propWithProps.nestedPropD\` | () => \`boolean\` | - |
"
`;

exports[`Interface Reflection should compile extended interface: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Interface: ExtendedInterface

Comments for ExtendedInterface

## Extends

- [\`BasicInterface\`](BasicInterface.md)

## Contents

- [Properties](ExtendedInterface.md#properties)
    - [deprecatedProp](ExtendedInterface.md#deprecatedprop)
    - [extendedProp](ExtendedInterface.md#extendedprop)
    - [functionProp](ExtendedInterface.md#functionprop)
    - [optionalProp](ExtendedInterface.md#optionalprop)
    - [prop](ExtendedInterface.md#prop)
    - [propWithProps](ExtendedInterface.md#propwithprops)

## Properties

### ~~deprecatedProp~~

> **deprecatedProp**: \`string\`

#### Deprecated

This prop is deprecte

#### Some Tag

Comments for some tag

#### Inherited from

[\`BasicInterface.deprecatedProp\`](BasicInterface.md#deprecatedprop)

#### Source

[interfaces.ts:25](http://source-url)

***

### extendedProp

> **extendedProp**: \`string\`

#### Source

[interfaces.ts:84](http://source-url)

***

### functionProp

> **functionProp**: (\`s\`) => \`boolean\`

Comments for functionProper

#### Parameters

• **s**: \`string\`

Comment for param s

#### Returns

\`boolean\`

#### Inherited from

[\`BasicInterface.functionProp\`](BasicInterface.md#functionprop)

#### Source

[interfaces.ts:36](http://source-url)

***

### optionalProp?

> **optionalProp**?: \`string\`

Comments for optional prop

#### Inherited from

[\`BasicInterface.optionalProp\`](BasicInterface.md#optionalprop)

#### Source

[interfaces.ts:30](http://source-url)

***

### prop

> **prop**: \`string\`

Comments for prop

#### Inherited from

[\`BasicInterface.prop\`](BasicInterface.md#prop)

#### Source

[interfaces.ts:14](http://source-url)

***

### propWithProps

> **propWithProps**: \`Object\`

Comments for propWithProps

#### Type declaration

##### callbacks?

> **callbacks**?: \`Partial\`\\<[\`CallbacksOptions\`](../classes/CallbacksOptions.md)\\<[\`DisposableClass\`](../classes/DisposableClass.md), [\`ClassWithModifiers\`](../classes/ClassWithModifiers.md)\\>\\>

Comments for callbacks

##### nestedPropA

> **nestedPropA**: \`string\`

Comments for nestedPropA

##### nestedPropB

> **nestedPropB**: \`boolean\`

Comments for nestedPropB

##### nestedPropC

> **nestedPropC**: \`Object\`

Comments for nestedPropC

##### nestedPropC.nestedPropCA

> **nestedPropC.nestedPropCA**: \`string\`

Comments for nestedPropCA

##### nestedPropD

> **nestedPropD**: () => \`boolean\`

Comments for nestedPropD

###### Returns

\`boolean\`

#### Inherited from

[\`BasicInterface.propWithProps\`](BasicInterface.md#propwithprops)

#### Source

[interfaces.ts:41](http://source-url)
"
`;

exports[`Interface Reflection should compile extended interface: (Output File Strategy "members") (Option Group "2") 1`] = `
"# ExtendedInterface

Comments for ExtendedInterface

## Extends

- [\`BasicInterface\`](BasicInterface.md)

## Contents

- [Properties](ExtendedInterface.md#properties)

## Properties

| Property | Type | Description | Inheritance |
| :------ | :------ | :------ | :------ |
| ~~\`deprecatedProp\`~~ | \`string\` | **Deprecated**<br />This prop is deprecte<br /><br />**Some Tag**<br />Comments for some tag | [\`BasicInterface.deprecatedProp\`](BasicInterface.md) |
| \`extendedProp\` | \`string\` | - | - |
| \`functionProp\` | (\`s\`) => \`boolean\` | - | [\`BasicInterface.functionProp\`](BasicInterface.md) |
| \`optionalProp?\` | \`string\` | Comments for optional prop | [\`BasicInterface.optionalProp\`](BasicInterface.md) |
| \`prop\` | \`string\` | Comments for prop | [\`BasicInterface.prop\`](BasicInterface.md) |
| \`propWithProps\` | \\{   \`callbacks\`: \`Partial\`\\<[\`CallbacksOptions\`](../classes/CallbacksOptions.md)\\<[\`DisposableClass\`](../classes/DisposableClass.md), [\`ClassWithModifiers\`](../classes/ClassWithModifiers.md)\\>\\>;   \`nestedPropA\`: \`string\`;   \`nestedPropB\`: \`boolean\`;   \`nestedPropC\`: \\{      \`nestedPropCA\`: \`string\`;   };   \`nestedPropD\`: () => \`boolean\`;   } | Comments for propWithProps | [\`BasicInterface.propWithProps\`](BasicInterface.md) |
| \`propWithProps.callbacks?\` | \`Partial\`\\<[\`CallbacksOptions\`](../classes/CallbacksOptions.md)\\<[\`DisposableClass\`](../classes/DisposableClass.md), [\`ClassWithModifiers\`](../classes/ClassWithModifiers.md)\\>\\> | Comments for callbacks | - |
| \`propWithProps.nestedPropA\` | \`string\` | Comments for nestedPropA | - |
| \`propWithProps.nestedPropB\` | \`boolean\` | Comments for nestedPropB | - |
| \`propWithProps.nestedPropC\` | \\{   \`nestedPropCA\`: \`string\`;   } | Comments for nestedPropC | - |
| \`propWithProps.nestedPropC.nestedPropCA\` | \`string\` | Comments for nestedPropCA | - |
| \`propWithProps.nestedPropD\` | () => \`boolean\` | - | - |
"
`;

exports[`Interface Reflection should compile indexable interface: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Interface: IndexableInterface

Comments for IndexableInterface

## Indexable

 \\[\`s\`: \`string\`\\]: \`string\`

## Contents

- [Properties](IndexableInterface.md#properties)
    - [prop](IndexableInterface.md#prop)

## Properties

### prop

> **prop**: \`string\`

#### Source

[interfaces.ts:92](http://source-url)
"
`;

exports[`Interface Reflection should compile indexable interface: (Output File Strategy "members") (Option Group "2") 1`] = `
"# IndexableInterface

Comments for IndexableInterface

## Indexable

 \\[\`s\`: \`string\`\\]: \`string\`

## Contents

- [Properties](IndexableInterface.md#properties)

## Properties

| Property | Type | Description |
| :------ | :------ | :------ |
| \`prop\` | \`string\` | - |
"
`;

exports[`Interface Reflection should compile interface with type parameters: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Interface: InterfaceWithTypeParameters\\<A\\>

Comments for InterfaceWithTypeParameters

## Type parameters

• **A**

## Contents

- [Properties](InterfaceWithTypeParameters.md#properties)
    - [prop](InterfaceWithTypeParameters.md#prop)

## Properties

### prop

> **prop**: \`A\`

Comments for prop

#### Source

[interfaces.ts:77](http://source-url)
"
`;

exports[`Interface Reflection should compile interface with type parameters: (Output File Strategy "members") (Option Group "2") 1`] = `
"# InterfaceWithTypeParameters\\<A\\>

Comments for InterfaceWithTypeParameters

## Type parameters

| Parameter |
| :------ |
| \`A\` |

## Contents

- [Properties](InterfaceWithTypeParameters.md#properties)

## Properties

| Property | Type | Description |
| :------ | :------ | :------ |
| \`prop\` | \`A\` | Comments for prop |
"
`;
